### PR TITLE
[CS2103T-F13-2] $aveNUS

### PR DESCRIPTION
**Pull request by seanlowjk**
@jon-chua
@robytanama
@fatclarence
@Raikonen

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -361,14 +361,86 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
++
+Use case ends.
+
+[discrete]
+=== Use case: Display recommendations
+
+*MSS*
+
+1. User requests to list recommendations.
+2. $aveNUS shows a list of recommended food items based on their specified budget.
```
Comment (kerwin97) [inline docs/DeveloperGuide.adoc:1483]: For this use case, could there be a  possible extension to choose the budget when searching for recommendations? A user might have to keep setting budget if he/she would like to increase the search. 
It is difficult to provide possible extensions as we are not sure how your program will be like. 
Use Cases are all satisfactory, fulfilling the tutorial's requirements. Step numbering correct,
possible extensions mentioned, no unnecessary UI details, no formatting/notional errors. Nicely done!

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -137,130 +146,245 @@ image:BetterModelClassDiagram.png[]
...
-image::CommitActivityDiagram.png[]
+The command is read as a text string from the command box in the UI and then is executed by calling MainWindow#executeCommand(),
+which passes this string (named commandText) to the Logic component by calling Logic#execute(Model model).
 
-==== Design Considerations
+The sequence diagram for interactions between the Logic, Model and Storage components when a user executes the `save` command is shown below.
 
-===== Aspect: How undo & redo executes
+.Sequence diagram for sample save command
+image::SaveSequenceDiagram.png[]
```
Comment (zhangxuan97) [inline docs/DeveloperGuide.adoc:506]: Technical details:
- For this save sequence diagram, perhaps you can consider removing the participant SavingsAccount as it does not seem to be used. 
- For LogicManager, the activation bar seems to cut off prematurely - before Storage#saveSavingsAccount (it should deactivate at the bottom arrow instead).
- Storage should be StorageManager instead, as Storage is an interface.

Additionally, maybe you can consider splitting this sequence diagram (put a reference frame over LogicManager, Storage, and JsonStorageManager and include the storage components in the separate diagram).

**Review by zhangxuan97 — COMMENTED** [review]
Generally, the developers guide is well written - our group feels that your recommend feature is very comprehensive. The sorting feature can be more elaborated on - perhaps you guys can make more reference to the class diagram that you drew.

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -295,35 +457,146 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+|`*` |user who loves to customize things |have a theme changing function of the app from a list of themes available | personalise the app to a theme that I like.
+
+|`*` |user who hates travelling | sort places from the nearest to the furthest from my current location| find food places that are easy for me to get to.
+
+|`*` |user |add the birthdays of my friends |set up reminders to buy gifts for my friends.
+
+|`*` |user |receive sample suggestions and examples to understand how to use the program. |understand how to user the application effectively.
+
+|`*` |forgetful user |save specific meal sets to reuse |save time on inputting my meals daily.
+|=======================================================================
```
Comment (TeoShyanJie) [inline docs/DeveloperGuide.adoc:1375]: Perhaps you could rearrange the user in alphabetical order to aid in understanding what a role can do.

**Pull request closed without merging**